lib/fiddle/truffleruby.rb

```diff
@@ -0,0 +1,623 @@
+# frozen_string_literal: true
+# truffleruby_primitives: true
+
+# Copyright (c) 2019, 2024 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 2.0, or
+# GNU General Public License version 2, or
+# GNU Lesser General Public License version 2.1.
+
+module Truffle::FiddleBackend
+
+  SIZEOF_CHAR   = Primitive.pointer_find_type_size(:char)
```

Comment:
`Primitive` might be renamed or changed, etc.
So far they were only used inside TruffleRuby so there was no compatibility needed whatsoever.
`Primitive`s so far have been an implementation detail (and in fact even the syntax to use them changed).

If a gem uses a Primitive then we need to make that Primitive stable.
One idea there is to annotate it in https://github.com/oracle/truffleruby (e.g. with `@Stable`, `stable = true`, `stable = "fiddle"` or `usedBy = "fiddle"`) so we know we shouldn't change its name or behavior.
And also make sure exposing that behavior as a Primitive makes sense or is necessary.

We should probably also expose a predicate to find out if a Primitive is defined or not, since the set of Primitives changes with different TruffleRuby versions. Maybe `TruffleRuby.primitive?(name)`.
That way if we want to use a new Primitive but it might not always be available it's possible.
And it's also nicer than comparing versions.

I don't want TruffleRuby to repeat the mistake of exposing all internal APIs as "public API because used in some gem" (which was the case for CRuby C API and for JRuby Java API).

Reply:
We can do it or we can use `ffi` API like JRuby does.